src/find/matchers/printer.rs

```diff
@@ -25,25 +27,44 @@ impl std::fmt::Display for PrintDelimiter {
 /// This matcher just prints the name of the file to stdout.
 pub struct Printer {
     delimiter: PrintDelimiter,
+    output_file: Option<File>,
 }
 
 impl Printer {
-    pub fn new(delimiter: PrintDelimiter) -> Self {
-        Self { delimiter }
+    pub fn new(delimiter: PrintDelimiter, output_file: Option<File>) -> Self {
+        Self {
+            delimiter,
+            output_file,
+        }
     }
 }
 
 impl Matcher for Printer {
     fn matches(&self, file_info: &DirEntry, matcher_io: &mut MatcherIO) -> bool {
-        let mut out = matcher_io.deps.get_output().borrow_mut();
-        write!(
-            out,
-            "{}{}",
-            file_info.path().to_string_lossy(),
-            self.delimiter
-        )
-        .unwrap();
-        out.flush().unwrap();
+        match &self.output_file {
+            Some(output_file) => {
+                let mut out = output_file;
+                write!(
+                    out,
+                    "{}{}",
+                    file_info.path().to_string_lossy(),
+                    self.delimiter
+                )
+                .unwrap();
```

Comment:
GNU find doesn't treat this as a fatal error.  You can try `-fprint /dev/full` on Linux for example.

Reply:
Thanks, the other comment in the current review have been revised.
and GNU find seems to also interrupt the traversal of the remaining files. Should we also interrupt the traversal and give a suitable message?

Reply:
Thanks, changed in new commit.